Update contract getting started

BlaineHeffron opened the pull request:
Updates the smart contracts getting started tutorial to reflect the new ContractClient class in the js-stellar-sdk. Also fixes a typo of the hello world contract name.

ElliotFriend reviewed (approved):
Thanks for the fixes! Looks great

chadoh commented on docs/smart-contracts/getting-started/create-an-app.mdx:
```diff
@@ -255,20 +255,21 @@ Now let's add a new component to the `src/components` directory called `ConnectF
 
   if (await isAllowed()) {
-    if (await getPk()) setLoggedIn();
+    const publicKey = await getPk();
+    if (publicKey) setLoggedIn(publicKey);
```
Why this change?

BlaineHeffron replied:
Change was to avoid the two await getUserInfo calls but in retrospect its cost is negligible so doesn't matter.

chadoh commented on docs/smart-contracts/getting-started/create-an-app.mdx:
```diff
@@ -324,23 +325,30 @@ Current value: <strong id="current-value" aria-live="polite">???</strong><br />
   const button = document.querySelector("[data-increment]");
   const currentValue = document.querySelector("#current-value");
-
+  async function getPk() {
+    const { publicKey } = await getUserInfo();
+    return publicKey;
+  }
+  if (await isAllowed()) {
+    const publicKey = await getPk();
+    if (publicKey) incrementor.options.publicKey = publicKey;
+  }
```
This is fine for now, but I think Freighter now has a `getPublicKey` or just `publicKey` method that we can use here to simplify this logic.

BlaineHeffron replied:
You are correct, this code works:

chadoh commented on docs/smart-contracts/getting-started/create-an-app.mdx:
```diff
@@ -324,23 +325,30 @@ Current value: <strong id="current-value" aria-live="polite">???</strong><br />
       '<span class="visually-hidden"> – updating…</span>';
 
-    const tx = await incrementor.increment();
+    const tx = await incrementor.increment({timeoutInSeconds: 30});
```
Do we need this? Should the default be increased from `10` so we don't need it anymore?

BlaineHeffron replied:
It seems to be needed, which appears to be a bug - https://github.com/stellar/js-stellar-sdk/issues/950

BlaineHeffron replied:
I took it out because the default will be changed to 60 for release.

chadoh reviewed (approved):
Looks good. I have a couple questions and notes, but none are show-stoppers. I think these are good changes that we should merge sooner rather than later, and then maybe follow up with more PRs later.

BlaineHeffron commented:
I added the `signTransaction` method to the sign and send in the `Counter` script. This is needed based on how the current soroban astro template creates the `Client`. I also updated the url to the new stellar repo for the soroban astro template. I removed the `timeoutInSeconds` passed to the increment because it shouldn't be needed with the new default being set to 60 (see https://github.com/stellar/js-stellar-sdk/pull/951). I simplified the logic for getting the public key in `Counter`.

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr520.previews.kube001.services.stellar-ops.com

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr520.previews.kube001.services.stellar-ops.com
